### PR TITLE
Switch away from ansible for packet devices

### PR DESCRIPTION
**Pull request by derekhiggins**
The packet ansible playbooks hide a lot of the errors and
are proving to be error prone, switch to using curl to
check the device status.

--
If doing this we may aswell stop using ansible for the device creation also...

### CODE REVIEW

File: ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh [hunk 1]
```diff
@@ -115,3 +95,32 @@ cat > packet-setup.yaml <<-EOF
...
 
 ansible-playbook packet-setup.yaml -e "packet_hostname=ipi-${NAMESPACE}-${JOB_NAME_HASH}-${BUILD_ID}"  |& gawk '{ print strftime("%Y-%m-%d %H:%M:%S"), $0; fflush(); }'
+
+DEVICEID=$(jq -r .devices[0].id < ${SHARED_DIR}/hosts.json)
+
+function refresh_device_info(){
+    curl -H "X-Auth-Token: $(cat ${CLUSTER_PROFILE_DIR}/packet-auth-token)"  "https://api.equinix.com/metal/v1/devices/$DEVICEID" > /tmp/device.json
+    STATE=$(jq -r .state < /tmp/device.json)
+    IP=$(jq -r .ip_addresses[0].address < /tmp/device.json)
+}
```
Comment (andfasano) [inline ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:105]: Implemention is fine, but just for the sake of consistency my preference would be to have them as Ansible task (I think the [uri](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/uri_module.html) module could be used for API calls, and also [shell](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/shell_module.html) could be useful)

Comment (derekhiggins) [reply ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:105]: I intentionally stayed away from ansible as we'd be more likely to see the raw errors with curl (i.e. ansible is hiding what actually happened from us)

Comment (andfasano) [reply ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:105]: We could add the `{{ hosts }}` to the slack reporting message also to fetch it

Comment (derekhiggins) [reply ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh:105]: I've added the contents of the json to the slack message but I'm also interested in seeing HTTP status codes so stuck to using curl for the moment.

**Comment by andfasano** [comment]
@derekhiggins I guess this one is not needed anymore?

**Comment by derekhiggins** [comment]
/test pj-rehearse

@andfasano @elfosardo 
Looking at the results for the Jobs in the last to rehearses on this, of the Jobs that requested a host from packet
5 got their host from OFCIR
7 got their host using the ansible/curl added in this PR
0 failed to get a host from packet

I've triggered another rehearse, if all looks ok I think its good to merge

File: ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-ref.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ref:
   as: baremetalds-packet-setup
   from: dev-scripts
+  grace_period: 1m0s
```
Comment (andfasano) [inline ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-ref.yaml:4]: What's this for?

Comment (derekhiggins) [reply ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-ref.yaml:4]: I guess if the pod is killed it to ensure the trap doesn't prevent it exiting indefinitely

**Comment by derekhiggins** [comment]
> /test pj-rehearse
> 
> @andfasano @elfosardo Looking at the results for the Jobs in the last 2 rehearses on this, of the Jobs that requested a host from packet 5 got their host from OFCIR 7 got their host using the ansible/curl added in this PR 0 failed to get a host from packet
> 
> I've triggered another rehearse, if all looks ok I think its good to merge

ok, in the last reharsal, 10 jobs requested a host from packet, OFCIR was down  so they all fell back to 
requesting a on demand host and all 10 got one, thats 100% success rate for 17 on demand packet hosts.

**Pull request merged**